app/assets/javascripts/libs/utils.js

```diff
@@ -244,6 +244,51 @@ const Utils = {
   toNullable<T>(maybe: Maybe<T>): ?T {
     return maybe.isJust ? maybe.get() : null;
   },
+
+  getRecursiveKeysAndValues(obj: Object): Array<any> {
+    return _.flattenDeep(Utils.getRecursiveKeysAndValuesUnflat(obj));
+  },
+
+  getRecursiveKeysAndValuesUnflat(obj: Object): Array<any> {
+    if (_.isArray(obj)) {
+      return obj.map(Utils.getRecursiveKeysAndValuesUnflat);
+    } else if (_.isObject(obj)) {
+      return Object.keys(obj).map(key => [key, Utils.getRecursiveKeysAndValuesUnflat(obj[key])]);
+    } else {
+      return [obj];
+    }
+  },
+
+  // Filters an array given a search string. Supports regex seach and several words as OR query.
+  // Supports nested properties
+  filterWithSearchQuery<T: Object>(
+    collection: Array<T>,
+    properties: Array<string>,
+    searchQuery: string,
+  ): Array<T> {
+    if (searchQuery === "") {
+      return collection;
+    } else {
+      const words = _.map(searchQuery.split(" "), element =>
+        element.toLowerCase().replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&"),
+      );
+      const uniques = _.filter(_.uniq(words), element => element !== "");
+      const pattern = `(${uniques.join("|")})`;
+      const regexp = new RegExp(pattern, "igm");
```

Comment:
Whoops, right, I mixed it up. However, multiline is a bit superfluous, too, since we don't use `^` or `$` in the regex (won't matter performance wise, though).